win32ss/user/ntuser/defwnd.c

```diff
@@ -529,6 +529,122 @@ DefWndScreenshot(PWND pWnd)
     UserCloseClipboard();
 }
 
+static VOID UserWindowSnapByKey(WPARAM wParam, LPARAM lParam)
+{
+    BOOL IsTaskBar;
+    DWORD Style;
+    DWORD ExStyle;
+    HWND hwndTop = UserGetForegroundWindow();
+    PWND topWnd = UserGetWindowObject(hwndTop);
+    USER_REFERENCE_ENTRY TopRef;
+
+    if (!topWnd)
+        return;
+
+    UserRefObjectCo(topWnd, &TopRef);
+
+    /* Test for typical TaskBar ExStyle Values */
+    ExStyle = (topWnd->ExStyle & WS_EX_TOOLWINDOW);
+    /* Test for typical TaskBar Style Values */
+    Style = (topWnd->style & (WS_POPUP | WS_VISIBLE | WS_CLIPSIBLINGS | WS_CLIPCHILDREN));
+    /* Test for masked typical TaskBar Style and ExStyles to detect TaskBar */
+    IsTaskBar = (Style == (WS_POPUP | WS_VISIBLE | WS_CLIPSIBLINGS | WS_CLIPCHILDREN)) &&
+                (ExStyle == WS_EX_TOOLWINDOW);
+
+    TRACE("ExStyle=%x, Style=%x, IsTaskBar=%d\n", ExStyle, Style, IsTaskBar);
+
+    /* Second test is so we are not touching the Taskbar */
```

Comment:
The new comment is misleading. You mean the **first** test, not the second test. I do recommend to strip the comment entirely, because it is of Captain Obvious style.